packages/edge-gateway/src/ipfs.js

```diff
@@ -27,5 +27,14 @@ export async function ipfsGet(request, env) {
     `https://${nCid}.${env.IPFS_GATEWAY_HOSTNAME}${redirectPath}${redirectQueryString}`
   )
 
-  return Response.redirect(url, 302)
+  const headers = new Headers(request.headers)
+  headers.set('Referrer-Policy', 'unsafe-url')
+  headers.set('Location', url.toString())
+  headers.set('Referer', request.headers.get('Referer'))
```

Comment:
Does this work IRL? I thought this was one of those headers you're not allowed to set in a browser/worker context!

Reply:
Also, this what we send back to the client right? `Referer` is a request header, I'm not sure I understand how this helps?

Reply:
This fix is based on discussion https://community.cloudflare.com/t/passing-on-referer-value-on-301-https-redirects/78787/8

The goal is to have CF analytics / Firewall have access to the Referer, which looks like not the case with current redirect. 

> Does this work IRL? 

I need to deploy to see, using staging dedicated route it does not break anything, but for actually seeing analytics in nftstorage.link domain I need to get it there